tenacity/__init__.py

```diff
@@ -88,6 +88,8 @@
 if t.TYPE_CHECKING:
     import types
 
+    from typing_extensions import Self
```

Comment:
This is not in tenacity dependency list; you'll need to add it.

Reply:
I'm honestly not sure if it should be in runtime dependencies, mypy seems to define `typing_extensions>=4.6.0` in it's own dependencies, but I'm not entirely sure about IDEs/Editors and Pyright

Reply:
Ah I missed it's in only in `TYPE_CHECKING`. I guess I don't know either. We can try this way and see what it breaks? Anyhow we'll remove it when we drop Python < 3.11 🤷